### PR TITLE
Update Module 3.1 to use CloudFront KeyValueStore

### PR DESCRIPTION
**Pull request by piotrekwitkowski**
Description of changes:

- [x] modify CDK code and write functions
- [x] update module description in README
- [ ] update architecture diagram for the module 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: resources/module_3_1/viewer-request-function.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    const returningUser = Boolean(request.cookies['X-Experiment']);
+    console.log('X_Experiment_U: ' + (returningUser ? 'RETURNING USER' : 'NEW USER'));
+
+    // If already visited, parse the experiment value, else generate one
+    const experimentValue = returningUser ? parseInt(request.cookies["X-Experiment"].value) : Math.floor(Math.random() * 100);
+    console.log('EXPERIMENT VALUE: ' + experimentValue);
+
+    // Set the correct URI for the experiment
+    request.uri = experimentValue > config.THRESHOLD ? config.URI_A : config.URI_B;
+    console.log('X_Experiment_V: ' + request.uri);
```
Comment (cornelcroi) [inline resources/module_3_1/viewer-request-function.js]: You should log something like `console.log("X_Experiment_V " + (request.uri == '/index.html' ? 'A_VERSION' : 'B_VERSION'));` and make sure that the dashboard still works.

File: lib/module_3_1/ab_testing_module_3_1.ts [hunk 2]
```diff
@@ -17,96 +16,63 @@
...
+      keyValueStoreImportSourcePath: RESOURCES_PATH_PREFIX + "/viewer-request-store-source.json"
     });
 
-    const lambdaEdgeViewerRequest = new lambda.Function(this, 'LambdaEdgeViewerRequest', {
-      runtime: lambda.Runtime.NODEJS_16_X,
-      handler: 'index.handler',
-      code: lambda.Code.fromAsset('resources/module_3_1/request'),
-    });
+    const viewerResponse = new Function(this, "viewer-response", {
+      code: FunctionCode.fromFile({ filePath: RESOURCES_PATH_PREFIX + "/viewer-request-function.js" }),
```
Comment (cornelcroi) [inline lib/module_3_1/ab_testing_module_3_1.ts]: viewer response points to viewer request code

Comment (piotrekwitkowski) [reply lib/module_3_1/ab_testing_module_3_1.ts]: Fixed in https://github.com/piotrekwitkowski/ab-testing-at-edge/commit/672897e4d00bcd483e0d63ce904ff8208a2f4f68

File: resources/module_3_1/viewer-request-function.js (hunk 1 above)
Comment (piotrekwitkowski) [reply resources/module_3_1/viewer-request-function.js]: Fixed in https://github.com/aws-samples/ab-testing-at-edge/pull/11/commits/c443a190a0dd303a7c015dcde479c12b736e838c

**Review by cornelcroi — APPROVED** [review]

**Pull request merged**